corehq/apps/app_manager/static/app_manager/js/report-module.js

```diff
@@ -89,7 +89,96 @@ var ReportModule = (function () {
     }
 
-    function ReportConfig(report_id, display, availableReportIds, reportCharts, graph_configs, language) {
+    function FilterConfig(report_id, reportId, filterValues, reportFilters, changeSaveButton) {
+        var self = this;
+
+        this.reportFilters = JSON.parse(JSON.stringify(reportFilters)) || {};
+        for (var _id in this.reportFilters) {
+            for (var i = 0; i < this.reportFilters[_id].length; i++) {
+                var filter = this.reportFilters[_id][i];
+                if (_id == report_id && filterValues.hasOwnProperty(filter.slug)) {
+                    filter.selectedValue = filterValues[filter.slug];
+                    filter.selectedValue.doc_type = ko.observable(filter.selectedValue.doc_type);
+                } else {
+                    filter.selectedValue = {
+                        doc_type: ko.observable(null)
+                    };
+                }
+                var filterFields = [
+                    'custom_data_property',
+                    'date_range',
+                    'filter_type',
+                    'select_value'
+                ];
+                for(var filterFieldsIndex in filterFields) {
+                    filter.selectedValue[filterFields[filterFieldsIndex]] = ko.observable(filter.selectedValue[filterFields[filterFieldsIndex]] || '');
+                }
+                filter.selectedValue.value = ko.observable(filter.selectedValue.value ? filter.selectedValue.value.join("\u001F") : '');
+
+                filter.dynamicFilterName = ko.computed(function () {
+                    return reportId() + '/' + filter.slug;
+                });
+
+                if(filter.choices !== undefined && filter.show_all) {
+                    filter.choices.unshift({value: "_all", display: "Show All"}); // TODO: translate
+                }
+            }
+        }
+
+        this.selectedFilterStructure = ko.computed(function () { // for the chosen report
+            return self.reportFilters[reportId()];
+        });
+
+        this.toJSON = function () {
+            var selectedFilterStructure = self.selectedFilterStructure();
+            var selectedFilterValues = {};
+            for (var i = 0; i < selectedFilterStructure.length; i++) {
+                var filter = selectedFilterStructure[i];
+                if (filter.selectedValue.doc_type()) {
+                    selectedFilterValues[filter.slug] = {};
+                    selectedFilterValues[filter.slug].doc_type = filter.selectedValue.doc_type();
+                    // Depending on doc_type, pull the correct observables' values
+                    var docTypeToField = {
+                        AutoFilter: 'filter_type',
+                        CustomDataAutoFilter: 'custom_data_property',
+                        StaticChoiceFilter: 'select_value',
+                        StaticDatespanFilter: 'date_range'
+                    };
+                    for(var docType in docTypeToField) {
```

Comment:
this is the same iterating over an object problem from the other PR (also semi-funny indentation)